packages/good-design/src/apps/bridge/useBalanceHook.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+import { ethers } from "ethers";
+import { useEthers } from "@usedapp/core";
+import { IGoodDollar } from "@gooddollar/goodprotocol/types";
+import { useGetContract, useGetEnvChainId } from "@gooddollar/web3sdk-v2";
+import { Lock } from "async-await-mutex-lock";
+
+import { G$Amount } from "@gooddollar/web3sdk-v2";
+
+const pollLock = new Lock();
+
+const g$BalanceFromExplorer = async (account: string | undefined, contracts: string[]) => {
```

Comment:
why this hook? didnt we have a way to get balances from blockchain?

Reply:
Yes, but I did not want to use useDapp hooks (there almost no way to control their polling)

Reply:
what do you mean, we control their polling everywhere. in many of the hooks we have the polling interval

Reply:
yes which triggers around 8 re-renders

Reply:
Besides that, we only have one useG$Balance hook, which is for G$ / GOOD / GDX
which also fetches for mainnet.

It was either refactoring hook / adding new hook / using explorer.

I like the flow with the explorer more (There is no need to know the balance every x blocks)